qiskit/circuit/commutation_library.py

```diff
@@ -12,9 +12,9 @@
 
 """Provides a commutation checker that caches the determined commutation results during this session """
 
-from qiskit.circuit import CommutationChecker
```

Comment:
Should we make the Python commutation checker use the fast Rust implementation as well (until the former is removed)? 🙂 Or do you think this is unnecessary since they should just use this top-level import anyways?

Reply:
I don't know what the best way to handle this is... :-/ There is:
1. Removing the python commutation checker and let everything point to rust
2. Let the python commutation checker use rust internally
3. Leave it as it currently is

I think 1. may violate our stability policy but is probably something we want to converge to longterm. 2. Would yield the largest performance improvements but may yield strange results for users that are doing funky things with the python commutation checker that may not be supported by the Rust version. On the other hand this already happens with this version when users are invoking SessionCommutationChecker 3. does not have additional development overhead. I'm open to all these options. :-)

Reply:
Hmmm maybe 3 is indeed the best option together with a pending deprecation of the Python commutation checker?

Reply:
Oh, the commutation checker isn't actually public, it's not anywhere in the docs. so we don't have to be too worried about backwards compatibility on it

Reply:
Actually I might be wrong there, I see it in the docs for 1.0.2, but not for 1.2.0?

Reply:
If we use 1 or 2 we'd have to support the full caching of Parameters (in addition to only floats now) -- maybe we can add a flag for that if we see that it slows things down?

Reply:
This has been resolved and pattern 2 is in use now